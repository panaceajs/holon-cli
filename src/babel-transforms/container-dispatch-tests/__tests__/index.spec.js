const pluginTester = require('babel-plugin-tester');
const containerDispatchTests = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const pluginJsx = require('@babel/plugin-syntax-jsx');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: containerDispatchTests,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginJsx, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should change nothing when there is nothing to change.',
      fixture: '__fixtures__/with-tests.js',
      snapshot: false
    },
    {
      title: 'it should add test case for action types',
      fixture: '__fixtures__/with-tests.js',
      pluginOptions: { actionTypes: { first: 'first' } }
    },
    {
      title: 'it should not overwrite existing test case for action types',
      fixture: '__fixtures__/with-tests.js',
      pluginOptions: {
        actionTypes: { existingAction: 'existingActionValue', first: 'first' }
      }
    }
  ]
});
