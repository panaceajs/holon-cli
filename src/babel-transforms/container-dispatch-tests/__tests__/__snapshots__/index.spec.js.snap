// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-container-dispatch-tests it should add test case for action types: it should add test case for action types 1`] = `
"
describe('\`ExampleComponent\` component', () => {
  it('maps \`existingAction\` to dispatch \`EXISTING_ACTION\` action.', () => {
    // should not be modified by transform
    const existingActionValue = 'existingActionValue';
    const wrapper = shallow(<ExampleComponent store={store} />);
    wrapper.props().existingAction(existingActionValue);
    expect(store.dispatch).toHaveBeenCalledWith(existingAction(existingActionValue));

    // do other stuff
  });

  it('should be ignored by transform', () => {});

  it('does nothing');
});

describe('\`Some\` component');

      ↓ ↓ ↓ ↓ ↓ ↓

describe('\`ExampleComponent\` component', () => {
  it('maps \`existingAction\` to dispatch \`EXISTING_ACTION\` action.', () => {
    // should not be modified by transform
    const existingActionValue = 'existingActionValue';
    const wrapper = shallow(<ExampleComponent store={store} />);
    wrapper.props().existingAction(existingActionValue);
    expect(store.dispatch).toHaveBeenCalledWith(existingAction(existingActionValue));

    // do other stuff
  });

  it('should be ignored by transform', () => {});

  it('does nothing');
  it('maps \`first\` to dispatch \`FIRST\` action.', () => {
    const firstValue = 'firstValue';
    const wrapper = shallow(<ExampleComponent store={store} />);
    wrapper.props().first(firstValue);
    expect(store.dispatch).toHaveBeenCalledWith(first(firstValue));
  });
});

describe('\`Some\` component');
"
`;

exports[`holon-container-dispatch-tests it should not overwrite existing test case for action types: it should not overwrite existing test case for action types 1`] = `
"
describe('\`ExampleComponent\` component', () => {
  it('maps \`existingAction\` to dispatch \`EXISTING_ACTION\` action.', () => {
    // should not be modified by transform
    const existingActionValue = 'existingActionValue';
    const wrapper = shallow(<ExampleComponent store={store} />);
    wrapper.props().existingAction(existingActionValue);
    expect(store.dispatch).toHaveBeenCalledWith(existingAction(existingActionValue));

    // do other stuff
  });

  it('should be ignored by transform', () => {});

  it('does nothing');
});

describe('\`Some\` component');

      ↓ ↓ ↓ ↓ ↓ ↓

describe('\`ExampleComponent\` component', () => {
  it('maps \`existingAction\` to dispatch \`EXISTING_ACTION\` action.', () => {
    // should not be modified by transform
    const existingActionValue = 'existingActionValue';
    const wrapper = shallow(<ExampleComponent store={store} />);
    wrapper.props().existingAction(existingActionValue);
    expect(store.dispatch).toHaveBeenCalledWith(existingAction(existingActionValue));

    // do other stuff
  });

  it('should be ignored by transform', () => {});

  it('does nothing');
  it('maps \`first\` to dispatch \`FIRST\` action.', () => {
    const firstValue = 'firstValue';
    const wrapper = shallow(<ExampleComponent store={store} />);
    wrapper.props().first(firstValue);
    expect(store.dispatch).toHaveBeenCalledWith(first(firstValue));
  });
});

describe('\`Some\` component');
"
`;
